samcli/lib/package/artifact_exporter.py

```diff
@@ -257,65 +257,70 @@ class Resource:
     # Set this property to True in base class if you want the exporter to zip
     # up the file before uploading This is useful for Lambda functions.
     FORCE_ZIP = False
+    URL_PROTOCOL = "s3"
 
     def __init__(self, uploader, code_signer):
         self.uploader = uploader
         self.code_signer = code_signer
 
     def export(self, resource_id, resource_dict, parent_dir):
-        if resource_dict is None:
-            return
-
-        if resource_not_packageable(resource_dict):
-            return
-
-        property_value = jmespath.search(self.PROPERTY_NAME, resource_dict)
-
-        if not property_value and not self.PACKAGE_NULL_PROPERTY:
-            return
-
-        if isinstance(property_value, dict):
-            LOG.debug("Property %s of %s resource is not a URL", self.PROPERTY_NAME, resource_id)
-            return
-
+        context = {"resource_id": resource_id, "parent_dir": parent_dir}
+        _transform_dict(resource_dict, self.PROPERTY_NAME, self.parse_property, context)
+
+    def parse_property(self, original_dict, original_path, property_name, value, context):
+        if not value and not self.PACKAGE_NULL_PROPERTY:
+            return value
+        if isinstance(value, dict):
+            LOG.debug("Property %s of %s resource is not a URL", self.PROPERTY_NAME, context["resource_id"])
+            return value
         # If property is a file but not a zip file, place file in temp
         # folder and send the temp folder to be zipped
-        temp_dir = None
-        if is_local_file(property_value) and not is_zip_file(property_value) and self.FORCE_ZIP:
-            temp_dir = copy_to_temp_dir(property_value)
-            set_value_from_jmespath(resource_dict, self.PROPERTY_NAME, temp_dir)
-
+        temp_dir = False
+        if is_local_file(value) and not is_zip_file(value) and self.FORCE_ZIP:
+            temp_dir = True
+            value = copy_to_temp_dir(value)
```

Comment:
Variable name `value` is too generic here. Please update regarding to its function